File: 구름 알고리즘 미션/동적 프로그래밍/동전 줍기 대회/동전 줍기 대회.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const N = +input[0];
const coins = input[1].split(" ").map(Number);
let pickCoins = Array(N).fill(0);
pickCoins[0] = coins[0];

for(let i = 1 ; i < N ; i++){
	pickCoins[i] = Math.max(coins[i], coins[i] + pickCoins[i-1]);
}

console.log(Math.max(...pickCoins) > 0 ? Math.max(...pickCoins) : 0);

File: 구름 알고리즘 미션/동적 프로그래밍/동전 줍기 대회/동전 줍기 대회.py
import sys
input = sys.stdin.readline

N = int(input())
coins = list(map(int, input().split()))

pickCoins = [0] * N
pickCoins[0] = coins[0]
for i in range(1, N):
		pickCoins[i] = max(coins[i], coins[i] + pickCoins[i-1])

print(max(pickCoins) if max(pickCoins) > 0 else 0)